### PR TITLE
MGMT-13710: assisted-installer-controller hangs for 2 minutes before uploading the ingress CA

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-13710](https://issues.redhat.com//browse/MGMT-13710): assisted-installer-controller hangs for 2 minutes before uploading the ingress CA

Optimizing WaitForPredicateWithTimer first to run function and only then wait for interval.
Ingress CA upload should retry quickly.
Optimizing wait for operator is available by rechecking operator status in service right after update operator status to available

### CODE REVIEW

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: src/assisted_installer_controller/assisted_installer_controller_test.go [hunk 1]
```diff
@@ -1927,38 +1924,14 @@ var _ = Describe("installer HostRoleMaster role", func() {
...
+			mockbmclient.EXPECT().UpdateClusterOperator(gomock.Any(), gomock.Any(), cvoOperatorName, "", gomock.Any(), gomock.Any()).Return(fmt.Errorf("dummy")).AnyTimes()
 
 			err := func() error {
 				return assistedController.waitingForClusterOperators(ctx)
 			}()
 
 			Expect(errors.Is(err, context.DeadlineExceeded)).To(BeTrue())
 		})
-
-		It("service fail to sync - finally succeed", func() {
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller_test.go:1939]: Why did you remove this test?

File: src/inventory_client/inventory_client.go [hunk 2]
```diff
@@ -74,12 +74,6 @@ type HostData struct {
 	Host      *models.Host
 }
 
-func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, insecure bool, caPath string,
```
Comment (eranco74) [inline src/inventory_client/inventory_client.go:77]: dead code?

File: src/assisted_installer_controller/assisted_installer_controller_test.go (hunk 1 above)
Comment (tsorya) [reply src/assisted_installer_controller/assisted_installer_controller_test.go:1939]: Cause this test case actually tested that we do one more loop and not returning ready immediately

File: src/inventory_client/inventory_client.go (hunk 2 above)
Comment (tsorya) [reply src/inventory_client/inventory_client.go:77]: Yep, totally.
